src/helpers.rs

```diff
@@ -348,4 +349,42 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         Ok(())
     }
+
+    fn read_os_string(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
```

Comment:
One thing I just realized is that the strings we are reading here are *not* Rust `OsString`; they are "whatever the Unix APIs take as arguments" and they *turn that into* an `OsString`. We should find a name to reflect that in the name; this is important because if anyone used this function on a Windows API it would fail awkwardly.

The Unix APIs take as arguments null-terminated sequences of bytes, and that's why this function works.

Reply:
I have two candidates: `read_unix_string` or `read_os_string_from_unix_bytes`.

Reply:
I vote for a variant of the latter: `read/write_os_str_from/to_c_str`.